posts/2018-12-14-Rust-2018-dev-tools.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title: "Tools in the 2018 edition"
+author: Nick Cameron
+---
+
+Tooling is an important part of what makes a programming language practical and productive. Rust has always had some great tools (Cargo in particular has a well-deserved reputation as a best-in-class package manager and build tool), the 2018 edition includes more tools which we hope further improve Rust users' experience.
+
+In this blog post I'll cover Clippy and Rustfmt - two tools that have been around for a few years and are now stable and ready for general use. I'll cover IDE support - a key workflow for many users which is now much better supported. I'll start by talking about Rustfix, a new tool which was central to our edition migration plans.
+
+## Rustfix
+
+Rustfix is a tool for automatically making changes to Rust code. It is a key part of our migration story for the 2018 edition, making the transition from 2015 to 2018 editions much easier, and in many cases completely automatic. This is essential, since without such a tool we'd be much more limited in the kinds of breaking changes users would accept.
+
+A simple example:
+
+```rust
+trait Foo {}
+
+fn foo(&self, f: &Foo);
+```
+
+The above is legal in Rust 2015, but not in Rust 2018 (trait objects must be made explicit using `dyn`). Rustfix changes the above code to:
```

Comment:
Is it true that "trait objects must be made explicit using dyn"? 2018 seemed to accept trait objects without dyn when I tried it.